lighthouse-core/gather/gatherers/dobetterweb/domnodes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers all DOM nodes on the page.
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+class AllDOMNodes extends Gatherer {
+  /**
+   * @param {!Object} options
+   * @return {!Promise<!Array<!Object>>}
+   */
+  afterPass(options) {
+    return options.driver.querySelectorAll('body, body /deep/ *');
```

Comment:
agree with @patrickhulce that this may be a really big thing to bring back over and keep around just for the length. It doesn't look like the array contents would be particularly helpful for others audits, just an array of node IDs that would then each be wrapped by [Elements](https://github.com/GoogleChrome/lighthouse/blob/3043cc3732bd6b9cfbc02d96579a6f21c86de08b/lighthouse-core/lib/element.js), but we wouldn't be able to tell them apart in any way without more protocol requests for each one...

Particularly if you're going to have to get tricky with a selector for Shadow DOM v1, maybe switch to `evaluateAsync` and ship back only the number of nodes?